### ISSUE TITLE
Add "formality" parameter to DeepL

### ISSUE THREAD

**Issue by nilovelez** [vibgyj/WPTranslationFiller#321]
"Formality" parameter is needed to force formal/informal language, which important in some locales like es_ES

Reference:
https://www.deepl.com/docs-api/translate-text

**Comment by psmits1567** [comment]
Currently the addon detects if you are in the formal translation URL "formal", and sends the request with this parameter to DeepL.
Does es_ES has a "formal" URL?
 If so does the result of the request not contain the formal translation?
It would require some work to implement forcing formal requests for locales that do not have a 'formal' URL but still want formal tone from DeepL. Be aware this is only for a few languages available for DeepL

**Comment by nilovelez** [comment]
In es_ES we only use informal tone, so it would be ok if you just hardcoded it in the calls.
It is probably the same for oher locales, but you would have to check each one.

**Comment by psmits1567** [comment]
As indicated the add-on determines what tone it should use depending on the URL used. So if only informal tone is used, then there would be no need to force anything. The tag for DeepL is currently set to "formality=more" while the tag for informal is set to "formality=less" That is working fine for locale nl_NL.
So are you telling that the current version still uses formal translations?
If so than the only option we have is to use the setting "prefer_less" which is new to me.
Otherwise you have the option to replace single formal words from within the Postreplace list
Example
##form: je , u 
In this case after the translation comes back with 'je' which is informal, you can replace that by 'u' which is formal
The other way around you can do this
##def: u , je 
Here we get 'u' which is formal back, but is should be 'je' which is informal

**Comment by nilovelez** [comment]
Yes, in es_ES is using formal translation, which is a problem because in Spanish the sentence struture changes from formal to informal.

As I said, it would be enough to hardcode the "formality=less" tag to Locales that use informal by default.

**Comment by psmits1567** [comment]
That is already in place, so you should get 'informal' results. However DeepL can be stubborn sometimes as it uses stored translations. Hence the option with postreplace verbs to correct that.
Example the following URL is informal
https://translate.wordpress.org/projects/wp-plugins/jetpack-boost/stable-readme/nl/default/?filters%5Bstatus%5D=untranslated
Example the following is formal
https://translate.wordpress.org/projects/wp-plugins/jetpack-boost/stable-readme/nl/formal/?filters%5Bstatus%5D=untranslated
That structure is recognised, so if you have 'default' in your link it sends "formality=less'

**Comment by nilovelez** [comment]
It is using formal in all my tests. For example all this page is being translated as formal:
https://translate.wordpress.org/projects/wp-plugins/amp/stable/es/default/?filters%5Bstatus%5D=untranslated

**Comment by psmits1567** [comment]
Ok could you test it again with the latest branch 'improve_validate_page'
You need to install it in developer mode. Now it has a toggle to set whether you want to force it into 'formal' or not
If you do not set the toggle in options, it will handle the translations with 'formality=less'
If set it will force to translate into a formal tone with DeepL
Also can you provide an example of one line in 'informal' and in 'informal' tone please
My knowledge of Spanish is zero

**Comment by psmits1567** [comment]
Sorry it is 'formality=prefered_less'

**Comment by psmits1567** [comment]
Could you also indicate the formal words please

**Comment by psmits1567** [comment]
Can you test on this location to see what the result is?
https://www.deepl.com/docs-api/simulator

**Comment by nilovelez** [comment]
When I test using the extension published in the marketplace and I translate this string:

> Please check "Raw Data" for all error source information.

I always get this, which is formal ("Consulte", ending in e, as in "consulte usted"):

> Consulte "Datos brutos" para obtener toda la información sobre el origen de los errores.

Using the API simulator I only get that result when I set the formality flag to "default", "more" or "prefer_more"
If I set it to "less" or "prefer_less", I get:

> Consulta "Datos brutos" para obtener toda la información sobre el origen de los errores.

Which is informa ("Consulta" ending in a, as in "consulta tú")

Tomorrow I'll test the "improve_validate_page" branch

**Comment by psmits1567** [comment]
The extension in the marketplace is very old, so if you could test it with the branch indicated, I think the problem may be resolved. I will see if I can reproduce this issue. Although I have no idea why it does produce formal results.

**Comment by psmits1567** [comment]
I tested it with the latest version, and I get "Consulta" so it does give the informal translation.
Be aware that you need "CORS Unblock
0.3.7" to load a glossary for DeepL, otherwise you will get a "cross domain" error
